File: 251_Privacy.js
function solution(today, terms, privacies) {
  var answer = [];

  privacies.map((privacy, index) => {
    const [date, type] = privacy.split(" ");
    const [year, month, day] = date.split(".");
    const [tYear, tMonth, tDay] = today.split(".");

    const duration =
      Number(tYear - year) * 12 * 28 +
      Number(tMonth - month) * 28 +
      Number(tDay - day);

    terms.map((term) => {
      const [targetType, targetMonth] = term.split(" ");

      if (targetType === type && duration >= targetMonth * 28) {
        answer.push(index + 1);
      }
    });
  });
  return answer;
}

console.log(
  solution(
    "2020.01.01",
    ["Z 3", "D 5"],
    [
      "2019.01.01 D",
      "2019.11.15 Z",
      "2019.08.02 D",
      "2019.07.01 D",
      "2018.12.28 Z",
    ]
  )
);

File: 251_Privacy/251_Privacy.py
def solution(today, terms, privacies):
    tYear, tMonth, tDay = map(int, today.split("."))
    answer = []
    for i in range(len(privacies)):
        target, type = privacies[i].split(" ")
        year, month, day = map(int, target.split("."))

        term = (tYear - year) * 12 * 28 + (tMonth - month) * 28 + (tDay - day)
        for j in range(len(terms)):
            tarType, tarTerm = terms[j].split(" ")
            if tarType == type:
                if term >= int(tarTerm) * 28:
                    answer.append(i + 1)
    return answer

print(solution("2022.05.19", ["A 6", "B 12", "C 3"],	["2021.05.02 A", "2021.07.01 B", "2022.02.19 C", "2022.02.20 C"]))